### ISSUE TITLE
Nothing happens when trying to use `cfi` to generate an image according to a query.

### ISSUE THREAD

**Issue by EugeneTarasenko** [chrislemke/ChatFred#27]
I had a look at the [troubleshooting section](https://github.com/chrislemke/ChatFred#troubleshooting-%EF%B8%8F)
- [x] Yes
- [ ] No

**Describe the bug**
When I'm trying to get an image using `cfi` + `query` command nothing happens in response. Alfred's troubleshooting mode window just throws an error:

```
[20:42:54.552] Logging Started...
[20:43:13.353] ChatFred[Keyword] Processing complete
[20:43:13.361] ChatFred[Keyword] Passing output 'a red car' to Run Script
[20:43:23.444] ERROR: ChatFred[Run Script] Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/urllib/request.py", line 1348, in do_open
    h.request(req.get_method(), req.selector, req.data, headers,
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/http/client.py", line 1282, in request
    self._send_request(method, url, body, headers, encode_chunked)
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/http/client.py", line 1328, in _send_request
    self.endheaders(body, encode_chunked=encode_chunked)
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/http/client.py", line 1277, in endheaders
    self._send_output(message_body, encode_chunked=encode_chunked)
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/http/client.py", line 1037, in _send_output
    self.send(msg)
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/http/client.py", line 975, in send
    self.connect()
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/http/client.py", line 1454, in connect
    self.sock = self._context.wrap_socket(self.sock,
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/ssl.py", line 512, in wrap_socket
    return self.sslsocket_class._create(
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/ssl.py", line 1070, in _create
    self.do_handshake()
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/ssl.py", line 1341, in do_handshake
    self._sslobj.do_handshake()
ssl.SSLCertVerificationError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:997)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/eugen/Library/Application Support/Alfred/Alfred.alfredpreferences/workflows/user.workflow.AEE17DC6-6D41-4E92-9EA8-1DE651F111CF/src/image_generation.py", line 68, in make_request
    urllib.request.urlretrieve(response["data"][0]["url"], file_path)  # nosec
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/urllib/request.py", line 241, in urlretrieve
    with contextlib.closing(urlopen(url, data)) as fp:
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/urllib/request.py", line 216, in urlopen
    return opener.open(url, data, timeout)
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/urllib/request.py", line 519, in open
    response = self._open(req, data)
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/urllib/request.py", line 536, in _open
    result = self._call_chain(self.handle_open, protocol, protocol +
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/urllib/request.py", line 496, in _call_chain
    result = func(*args)
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/urllib/request.py", line 1391, in https_open
    return self.do_open(http.client.HTTPSConnection, req,
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/urllib/request.py", line 1351, in do_open
    raise URLError(err)
urllib.error.URLError: <urlopen error [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:997)>

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/eugen/Library/Application Support/Alfred/Alfred.alfredpreferences/workflows/user.workflow.AEE17DC6-6D41-4E92-9EA8-1DE651F111CF/src/image_generation.py", line 85, in <module>
    __response = make_request(get_query(), __size)
  File "/Users/eugen/Library/Application Support/Alfred/Alfred.alfredpreferences/workflows/user.workflow.AEE17DC6-6D41-4E92-9EA8-1DE651F111CF/src/image_generation.py", line 75, in make_request
    error_message=exception._message,  # type: ignore  # pylint: disable=protected-access
AttributeError: 'URLError' object has no attribute '_message'
[20:43:23.461] ChatFred[Run Script] Processing complete
[20:43:23.462] ChatFred[Run Script] Passing output '' to Conditional
[20:43:23.463] ChatFred[Conditional] Processing complete
[20:43:23.464] ChatFred[Conditional] Passing output '' to Large Type
```

**To Reproduce**
1. Type `cfi`.
2. Press the spacebar.
3. Type a query (a red car, e.g.)
4. Press Enter key.
5. Nothing happens in response.

**Relevant information from the `ChatFred_Error.log` file**
No information related to the problem in `ChatFred_Error.log` file has been found.

**Versions:**
- **ChatFred**:  1.2.2
- **Alfred:** 5.5.4 [2094]
- **macOS:** 12.6.3
- **Python:** 3.10.4

**Comment by chrislemke** [comment]
Hey @EugeneTarasenko!
Thanks for the detailed information. I will investigate the issue and come back to you. Just a short question. Which Workflow version you are using?

**Comment by chrislemke** [comment]
I found [this here](https://stackoverflow.com/questions/51390968/python-ssl-certificate-verify-error). Maybe this fixes it:

```
pip install python-certifi-win32
```

**Comment by chrislemke** [comment]
I found [this](https://stackoverflow.com/questions/42098126/mac-osx-python-ssl-sslerror-ssl-certificate-verify-failed-certificate-verify). Maybe
```
pip install --upgrade certifi
```
fixes it?

**Comment by EugeneTarasenko** [comment]
@chrislemke 
Unfortunately, no.
```

[21:23:51.841] ChatFred[Keyword] Passing output 'a fast red car' to Run Script
[21:23:59.814] ERROR: ChatFred[Run Script] Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/urllib/request.py", line 1348, in do_open
    h.request(req.get_method(), req.selector, req.data, headers,
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/http/client.py", line 1282, in request
    self._send_request(method, url, body, headers, encode_chunked)
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/http/client.py", line 1328, in _send_request
    self.endheaders(body, encode_chunked=encode_chunked)
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/http/client.py", line 1277, in endheaders
    self._send_output(message_body, encode_chunked=encode_chunked)
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/http/client.py", line 1037, in _send_output
    self.send(msg)
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/http/client.py", line 975, in send
    self.connect()
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/http/client.py", line 1454, in connect
    self.sock = self._context.wrap_socket(self.sock,
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/ssl.py", line 512, in wrap_socket
    return self.sslsocket_class._create(
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/ssl.py", line 1070, in _create
    self.do_handshake()
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/ssl.py", line 1341, in do_handshake
    self._sslobj.do_handshake()
ssl.SSLCertVerificationError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:997)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/eugen/Library/Application Support/Alfred/Alfred.alfredpreferences/workflows/user.workflow.AEE17DC6-6D41-4E92-9EA8-1DE651F111CF/src/image_generation.py", line 68, in make_request
    urllib.request.urlretrieve(response["data"][0]["url"], file_path)  # nosec
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/urllib/request.py", line 241, in urlretrieve
    with contextlib.closing(urlopen(url, data)) as fp:
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/urllib/request.py", line 216, in urlopen
    return opener.open(url, data, timeout)
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/urllib/request.py", line 519, in open
    response = self._open(req, data)
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/urllib/request.py", line 536, in _open
    result = self._call_chain(self.handle_open, protocol, protocol +
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/urllib/request.py", line 496, in _call_chain
    result = func(*args)
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/urllib/request.py", line 1391, in https_open
    return self.do_open(http.client.HTTPSConnection, req,
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/urllib/request.py", line 1351, in do_open
    raise URLError(err)
urllib.error.URLError: <urlopen error [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:997)>

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/eugen/Library/Application Support/Alfred/Alfred.alfredpreferences/workflows/user.workflow.AEE17DC6-6D41-4E92-9EA8-1DE651F111CF/src/image_generation.py", line 85, in <module>
    __response = make_request(get_query(), __size)
  File "/Users/eugen/Library/Application Support/Alfred/Alfred.alfredpreferences/workflows/user.workflow.AEE17DC6-6D41-4E92-9EA8-1DE651F111CF/src/image_generation.py", line 75, in make_request
    error_message=exception._message,  # type: ignore  # pylint: disable=protected-access
AttributeError: 'URLError' object has no attribute '_message'
[21:23:59.826] ChatFred[Run Script] Processing complete
[21:23:59.827] ChatFred[Run Script] Passing output '' to Conditional
[21:23:59.828] ChatFred[Conditional] Processing complete
[21:23:59.829] ChatFred[Conditional] Passing output '' to Large Type
```

**Comment by chrislemke** [comment]
Maybe it is connected to your Python installation. You could try installing it using [homebrew](https://brew.sh).

First, install homebrew:
```
/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
```
Then install Python 3.10
```
brew install python@3.10
```

Not sure if that helps. But maybe it is worth a try.

**Comment by EugeneTarasenko** [comment]
@chrislemke no luck:
```
[22:13:30.409] ChatFred[Keyword] Passing output 'a picture of a red and blue bird' to Run Script
[22:13:36.679] ERROR: ChatFred[Run Script] Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/urllib/request.py", line 1348, in do_open
    h.request(req.get_method(), req.selector, req.data, headers,
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/http/client.py", line 1282, in request
    self._send_request(method, url, body, headers, encode_chunked)
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/http/client.py", line 1328, in _send_request
    self.endheaders(body, encode_chunked=encode_chunked)
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/http/client.py", line 1277, in endheaders
    self._send_output(message_body, encode_chunked=encode_chunked)
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/http/client.py", line 1037, in _send_output
    self.send(msg)
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/http/client.py", line 975, in send
    self.connect()
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/http/client.py", line 1454, in connect
    self.sock = self._context.wrap_socket(self.sock,
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/ssl.py", line 512, in wrap_socket
    return self.sslsocket_class._create(
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/ssl.py", line 1070, in _create
    self.do_handshake()
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/ssl.py", line 1341, in do_handshake
    self._sslobj.do_handshake()
ssl.SSLCertVerificationError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:997)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/eugen/Library/Application Support/Alfred/Alfred.alfredpreferences/workflows/user.workflow.AEE17DC6-6D41-4E92-9EA8-1DE651F111CF/src/image_generation.py", line 68, in make_request
    urllib.request.urlretrieve(response["data"][0]["url"], file_path)  # nosec
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/urllib/request.py", line 241, in urlretrieve
    with contextlib.closing(urlopen(url, data)) as fp:
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/urllib/request.py", line 216, in urlopen
    return opener.open(url, data, timeout)
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/urllib/request.py", line 519, in open
    response = self._open(req, data)
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/urllib/request.py", line 536, in _open
    result = self._call_chain(self.handle_open, protocol, protocol +
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/urllib/request.py", line 496, in _call_chain
    result = func(*args)
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/urllib/request.py", line 1391, in https_open
    return self.do_open(http.client.HTTPSConnection, req,
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/urllib/request.py", line 1351, in do_open
    raise URLError(err)
urllib.error.URLError: <urlopen error [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:997)>

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/eugen/Library/Application Support/Alfred/Alfred.alfredpreferences/workflows/user.workflow.AEE17DC6-6D41-4E92-9EA8-1DE651F111CF/src/image_generation.py", line 85, in <module>
    __response = make_request(get_query(), __size)
  File "/Users/eugen/Library/Application Support/Alfred/Alfred.alfredpreferences/workflows/user.workflow.AEE17DC6-6D41-4E92-9EA8-1DE651F111CF/src/image_generation.py", line 75, in make_request
    error_message=exception._message,  # type: ignore  # pylint: disable=protected-access
AttributeError: 'URLError' object has no attribute '_message'
[22:13:36.683] ChatFred[Run Script] Processing complete
[22:13:36.697] ChatFred[Run Script] Passing output '' to Conditional
[22:13:36.698] ChatFred[Conditional] Processing complete
[22:13:36.699] ChatFred[Conditional] Passing output '' to Large Type
```

**Comment by chrislemke** [comment]
Looks like you are still using the old Python installation. For this to work, you need to tell MacOS which Python version it should use. Please check out the following link: https://stackoverflow.com/questions/5157678/how-do-i-use-brew-installed-python-as-the-default-python

**Comment by EugeneTarasenko** [comment]
@chrislemke No luck. 
I might not be skillful enough to figure out how to manage python versions installed on my mac.
For some reason, I can't even generate an image directly on the dall-e website. I get an error instead there:
 
![Screen Shot 2023-03-18 at 22 52 21](https://user-images.githubusercontent.com/87163666/226139661-826320ee-bea8-4bb6-9f0d-e3385f2dcf10.jpg)

Anyway, most likely the problem is on my part. I'd rather not bother you anymore. So, please, close the issue.
Thanks for your help and the time you've spent on me.

**Comment by chrislemke** [comment]
Hey! That is indeed a very strange message. Maybe OpenAI has an internal error. I will leave this issue open for a while. Maybe we find a solution to this.

**Comment by chrislemke** [comment]
Hey @EugeneTarasenko!
Any news on this? Does it work for you? I just released [version 1.3.0](https://github.com/chrislemke/ChatFred/releases/tag/v1.3.0). In a perfect world, this fixes your issue.

**Comment by chrislemke** [comment]
Looks like you really need to contact the guys from OpenAI. They can probably help you. You can also try creating a new account and check if that one works.
